src/pytom_tm/io.py

```diff
@@ -481,3 +482,103 @@ def read_defocus_file(file_name: pathlib.Path) -> list[float, ...]:
+        path to the tomogram for template matching; we use the name to pattern match in
+        the RELION5 star file
+    phase_flip_correction: bool, default False
+    phase_shift: float, default 0.0
+
+    Returns
+    -------
+    tomogram_voxel_size, tilt_angles, dose_accumulation, ctf_params, defocus_handedness:
+        tuple[float, list[float, ...], list[float, ...], list[dict, ...], int]
+    """
+    tomogram_id = tomogram_path.stem
+    tomograms_star_data = starfile.read(tomograms_star_path)
+
+    # match the tomo_id and check if viable
+    matches = [
+        (i, x)
+        for i, x in enumerate(tomograms_star_data["rlnTomoName"])
+        if x in tomogram_id
+    ]
+    if len(matches) == 1:
+        tomogram_meta_data = tomograms_star_data.loc[matches[0][0]]
+    else:
+        raise ValueError(
+            "Multiple or zero matches of tomogram id in RELION5 STAR file. Aborting..."
+        )
+
+    # grab the path to tilt series star for tilt angles, defocus and dose
+    tilt_series_star_path = pathlib.Path(
+        tomogram_meta_data["rlnTomoTiltSeriesStarFile"]
+    )
+    # update the path to a location we can actually find from CD
+    tilt_series_star_path = tomograms_star_path.parent.joinpath("tilt_series").joinpath(
+        tilt_series_star_path.name
+    )
+    tilt_series_star_data = starfile.read(tilt_series_star_path)
+
+    # we extract tilt angles, dose accumulation and ctf params
+    # TODO We need to have an internal structure for tilt series meta data
+    tilt_angles = list(tilt_series_star_data["rlnTomoNominalStageTiltAngle"])
+    dose_accumulation = list(tilt_series_star_data["rlnMicrographPreExposure"])
+
+    # _rlnTomoName  # 1
+    # _rlnVoltage  # 2
+    # _rlnSphericalAberration  # 3
+    # _rlnAmplitudeContrast  # 4
+    # _rlnMicrographOriginalPixelSize  # 5
+    # _rlnTomoHand  # 6
+    # _rlnTomoTiltSeriesPixelSize  # 7
+    # _rlnTomoTiltSeriesStarFile  # 8
+    # _rlnTomoTomogramBinning  # 9
+    # _rlnTomoSizeX  # 10
+    # _rlnTomoSizeY  # 11
+    # _rlnTomoSizeZ  # 12
+    # _rlnTomoReconstructedTomogram  # 13
+    tomogram_voxel_size = float(
+        tomogram_meta_data["rlnTomoTiltSeriesPixelSize"]
+        * tomogram_meta_data["rlnTomoTomogramBinning"]
+    )
+    defocus_handedness = int(tomogram_meta_data["rlnTomoHand"])
+
+    ctf_params = [
+        {
+            "defocus": defocus * 1e-6,
+            "amplitude_contrast": tomogram_meta_data["rlnAmplitudeContrast"],
+            "voltage": tomogram_meta_data["rlnVoltage"] * 1e3,
+            "spherical_aberration": tomogram_meta_data["rlnSphericalAberration"] * 1e-3,
+            "flip_phase": phase_flip_correction,
+            "phase_shift_deg": phase_shift,  # Unclear to me where the phase
+            # shifts are annotated in RELION5 tilt series file
+        }
+        for defocus in (
+            tilt_series_star_data.rlnDefocusV + tilt_series_star_data.rlnDefocusU
+        )
+        / 2
+        * 1e-4
```

Comment:
Yea it is supposed be that (I have the units in meters). The value in the relion file are in Angstrom, hence the factor 1e-10. I will change it to do it in one go, because this reads weird.